book_maker/translator/chatgptapi_translator.py

```diff
@@ -36,6 +36,7 @@
     "gpt-4o-mini",
     "gpt-4o-mini-2024-07-18",
 ]
+CONTEXT_PARAGRAPH_LIMIT = 3
```

Comment:
why we use global var here since it can set by user.

Reply:
@yihong0618 

Thank you for your question about the global variable `CONTEXT_PARAGRAPH_LIMIT`.
This global variable is used because it serves as a default value across multiple classes in the system, particularly in the `EPUBBookLoader` class.

The `EPUBBookLoader` class initializes `context_paragraph_limit` with a default value of 0:

```
python
class EPUBBookLoader(BaseBookLoader):
def init(
self,
# ... other parameters ...
context_paragraph_limit=0,
):
# ...
```

When this value (0) is passed to the `ChatGPTAPI` class, it triggers the use of the global `CONTEXT_PARAGRAPH_LIMIT` as a fallback:


```
python
if context_paragraph_limit > 0:
self.context_paragraph_limit = context_paragraph_limit
else:
self.context_paragraph_limit = CONTEXT_PARAGRAPH_LIMIT
```


This design allows for a consistent default value across the application while still providing flexibility for users to override it when necessary. It also maintains backwards compatibility with existing code that may not explicitly set this parameter.

That being said, if you have any suggestions for a better implementation that maintains the current functionality while reducing global variable usage, I'd be more than happy to consider and potentially implement it. I appreciate your keen eye for code quality and welcome any ideas that could enhance the codebase.